src/proc/interface.rs

```diff
@@ -145,16 +150,44 @@ impl<'a> Interface<'a> {
                             crate::Expression::AccessIndex { base, .. } => {
                                 left = base;
                             }
-                            crate::Expression::GlobalVariable(handle) => {
```

Comment:
as I'm thinking about this now, we don't want the visitor to repeat this descend themselves in order to reach that variable. How about this: instead of `visit_stmt` we have `visit_lhs_expr`, called on the final destination of this loop? It would mean "left-hand side" expression, and for the visitor it would mean that the IR wants to store/write into it.